interface/Device.tt

```diff
@@ -70,6 +70,30 @@ foreach (var register in deviceRegisters)
         };
     }
 
+    /// <summary>
+    /// Represents an operator that returns the contents of the metadata file
+    /// describing the <see cref="<#= deviceName #>"/> device registers.
+    /// </summary>
+    [Description("Returns the contents of the metadata file describing the <#= deviceName #> device registers.")]
+    public partial class DeviceMetadata : Source<string>
+    {
+        static readonly string Metadata = GetDeviceMetadata();
```

Comment:
Consider making this public so we can use in extensions if needed.

Reply:
This is a good idea, the only issue where I got stuck is how to name and expose the field:
- `Metadata` is not a good idea since then you would access it as `DeviceMetadata.Metadata` which is potentially misleading and ambiguous.
- `Instance` is a common name used for singleton classes and would read slightly better: `DeviceMetadata.Instance`. However, it might give the wrong impression that the entire class itself is a singleton, which is not the case since it is a regular operator class which you can create and place in the workflow.
- A final option is generating a third static class `Resources` which would contain the code to read the embedded metadata string: `Resources.DeviceMetadata`. The downside is adding more complication (another class) and potentially having to worry about name clashes with other custom project code.

Ideally we would keep the static readonly field in the class, but I haven't been able yet to come up with a good name for it. Perhaps `Value` might be a good choice?

Reply:
Following discussion with @bruno-f-cruz a better proposal is to move the static property to the `Device` class, where it will be accessible by writing `Device.Metadata`.

For clarity we have renamed the `DeviceMetadata` operator to `GetMetadata` which will be contextualized by the appropriate device namespace.